synapseclient/models/wiki.py

```diff
@@ -1 +1,1278 @@
+
+        if self.parent_id:
+            return "create_sub_wiki_page"
+
+        try:
+            await WikiHeader.get_async(
+                owner_id=self.owner_id, synapse_client=synapse_client
+            )
+        except SynapseHTTPError as e:
+            if e.response.status_code == 404:
+                return "create_root_wiki_page"
+            else:
+                raise
+        else:
+            if not self.id:
+                raise ValueError("Must provide id to update existing wiki page.")
+            return "update_existing_wiki_page"
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Store the wiki page: {self.owner_id}"
+    )
+    async def store_async(
+        self,
+        *,
+        synapse_client: Optional[Synapse] = None,
+    ) -> "WikiPage":
+        """Store the wiki page. If there is no wiki page, a new wiki page will be created.
+        If the wiki page already exists, it will be updated.
+        Subwiki pages are created by passing in a parent_id.
+        If a parent_id is provided, the wiki page will be created as a subwiki page.
+
+        Arguments:
+            synapse_client: If not passed in and caching was not disabled by
+                    `Synapse.allow_client_caching(False)` this will use the last created
+                    instance from the Synapse class constructor.
+
+        Returns:
+            The created/updated wiki page.
+
+        Raises:
+            ValueError: If owner_id is not provided or if required fields are missing.
+
+        Example: Store a wiki page
+            This example shows how to store a wiki page.
+            from synapseclient import Synapse
+            from synapseclient.models import (
+                Project,
+                WikiPage,
+            )
+            syn = Synapse()
+            syn.login()
+            project = await Project(name="My uniquely named project about Alzheimer's Disease").get_async()
+            wiki_page = await WikiPage(owner_id=project.id, title="My wiki page").store_async()
+            print(wiki_page)
+        """
+        client = Synapse.get_client(synapse_client=synapse_client)
+
+        wiki_action = await self._determine_wiki_action(synapse_client=client)
+        # get the markdown file handle and attachment file handles if the wiki action is valid
+        if wiki_action:
+            # Update self with the returned WikiPage objects that have file handle IDs set
+            self = await self._get_markdown_file_handle(synapse_client=client)
+            self = await self._get_attachment_file_handles(synapse_client=client)
+
+        # Handle root wiki page creation if parent_id is not given
+        if wiki_action == "create_root_wiki_page":
+            client.logger.info(
+                "No wiki page exists within the owner. Create a new wiki page."
+            )
+            # Create the wiki page
```

Comment:
nit: Lighten up on the number of in-line comments that don't expand beyond what the code is clearly completed. In this case `Create the wiki page`, doesn't add much information, but it adds a marginal maintenance burden to update it later.

IMO: Let the code speak for itself. If the code is complex enough that it needs an in-line comment, that is an indicator that the code could be structured a bit better to make it more clear.

Reply:
Cleaned the redundant comments.